inc/class-upgrade.php

```diff
@@ -157,6 +67,22 @@ public function __construct() {
 		do_action( 'wpseo_run_upgrade', $version );
 
 		$this->finish_up();
+
+	}
+
+	/**
+	 * Runs the upgrade routine.
+	 *
+	 * @param string $routine         The method to call.
+	 * @param string $version         The new version.
+	 * @param string $current_version The current set version.
+	 *
+	 * @return void
+	 */
+	protected function run_upgrade_routine( $routine, $version, $current_version ) {
+		if ( version_compare( $current_version, $version, '<' ) ) {
+			$this->$routine( $current_version );
```

Comment:
This function isn't covered by a unit test. But since most of this class isn't covered anyway I don't know whether or not you want to pick this up?